desktop/ui/src/main/java/org/datacleaner/user/MutableReferenceDataCatalog.java

```diff
@@ -167,18 +177,49 @@ public void removeDictionary(Dictionary dict, boolean externalize) {
             for (DictionaryChangeListener listener : _dictionaryListeners) {
                 listener.onRemove(dict);
             }
+            for(ReferenceDataChangeListener<Dictionary> listener : _dictionaryV2Listeners){
+                listener.onRemove(dict);
+            }
         }
         if (externalize) {
             _configurationWriter.removeDictionary(dict.getName());
             _userPreferences.save();
         }
     }
 
+    public void changeDictionary(Dictionary oldDictionary, Dictionary newDictionary){
+        changeDictionary(oldDictionary, newDictionary, true);
+    }
+    
+    public void changeDictionary(Dictionary oldDictionary, Dictionary newDictionary, boolean externalize) {
+        final List<Dictionary> dictionaries = _userPreferences.getUserDictionaries();
+        if (dictionaries.remove(oldDictionary)) {
+            if (externalize) {
+                _configurationWriter.removeDictionary(oldDictionary.getName());
+                _userPreferences.save();
+            }
+        }
+        
+        addDictionaryHelper(newDictionary, externalize);
+        for(ReferenceDataChangeListener<Dictionary> listener : _dictionaryV2Listeners){
+            listener.onChange(oldDictionary, newDictionary);
+        }
```

Comment:
Isn't this missing an update (remove+add) of the old-style listeners, or is that handled somewhere else (this goes for the string pattern as well).